source/common/stats/symbol_table_impl.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+#include <algorithm>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "envoy/common/exception.h"
+#include "envoy/stats/symbol_table.h"
+
+#include "common/common/assert.h"
+
+#include "absl/strings/str_join.h"
+#include "absl/strings/str_split.h"
+
+namespace Envoy {
+namespace Stats {
+
+class SymbolTableImpl : public SymbolTable {
+public:
+  std::vector<Symbol> encode(const std::string& name) override {
+    std::vector<Symbol> symbol_vec;
+    std::vector<std::string> name_vec = absl::StrSplit(name, '.');
```

Comment:
I think using string_view in this way isn't going to work -- `absl::string_view` doesn't implement hashable, so keeping an `std::unordered_map<absl::string_view, ...>` would require... a custom hasher which converts the string_view back into a string? Sadly I think we're stuck storing the full string in two locations.

Reply:
A custom hasher == "not a barrier" :)

It's pretty straightforward to do and definitely worth it to avoid extra allocations IMO.

Reply:
Won't this end up just being the cost of converting each string_view back into a string and taking the hash of that? Or is there a better way to write that custom hasher?

Reply:
No, see:

https://github.com/envoyproxy/envoy/blob/e89c9d6f23371971120e4c91119cdafe65d3a8df/source/common/stats/stats_impl.h#L174

and

https://github.com/envoyproxy/envoy/blob/3bf15a354ea1d0cc34d41571a59a7696d03872f7/source/common/common/utility.h#L490

Reply:
I think this would undo the performance benefits of only having to perform one map operation on `encode_map` (from elsewhere in this thread).

Since we store two maps:
``` 
encode_map_ <std::string, std::pair<Symbol, uint32_t>>;
decode_map_ <Symbol, std::string>;
```
we could replace the key in `encode_map_` with a string_view, but we would still have to store a full string as the value in the `decode_map_`, since we don't want to rely on the continued lifetime of some string outside of the SymbolTable. This would mean that the key in `encode_map_` should be a string_view into the value in the decode_map, which means that we have to perform a .find() on the encode_map before the .insert() into the encode_map.